kview/src/main/java/dev/ikm/komet/kview/controls/FilterOptionsUtils.java

```diff
@@ -620,11 +625,34 @@ public static <T> String getDescription(ViewCalculator viewCalculator, T t) {
             case String value -> value;
             case State value -> viewCalculator == null ?
                     Entity.getFast(value.nid()).description() :
-                    viewCalculator.getPreferredDescriptionTextOrNid(value.nid());
+                    getPreferredDescriptionTextOrNid(viewCalculator, value.nid());
             case Long value -> String.valueOf(value);
-            case EntityFacade value -> viewCalculator == null ?
-                    value.description() : viewCalculator.getPreferredDescriptionTextOrNid(value);
+            case EntityFacade value -> {
+                if (viewCalculator != null) {
+                    yield getPreferredDescriptionTextOrNid(viewCalculator, value);
+                }
+                yield value.description();
+            }
+
             default -> throw new RuntimeException("Unsupported type: " + t.getClass().getName());
         };
     }
+
+    public static String getPreferredDescriptionTextOrNid(ViewCalculator viewCalculator, int nid) {
+        try {
+            return viewCalculator.getPreferredDescriptionTextOrNid(nid);
```

Comment:
Can we change this to `.getDescriptionTextOrNid(nid)`?
reason: get preferred has an issue with always choosing regular name.